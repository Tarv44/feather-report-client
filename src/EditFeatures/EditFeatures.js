import React, {Component} from 'react';
import ProductContext from '../productContext';
import styles from './EditFeatures.module.css';
import { features, categories } from '../dummy-store';
import Feature from '../Feature/Feature';

export default class EditFeatures extends Component {
    static contextType = ProductContext;

    state = {
        catFeatures: []
    }

    componentDidMount() {
        const category = this.props.category
        const catFeatures = category !== 'Select Category' 
            ? features[this.props.category]
            : []
        this.setState({ catFeatures})
    }
    

    updateCatFeatures(category) {
        const catFeatures = features[category]
        this.setState({ catFeatures})
        this.props.handleCategory(category)
    }

    submitFeature = (feature) => {
        const catFeatures = this.state.catFeatures
        catFeatures.push({message: feature})
        this.setState({ catFeatures })
    }

    render() {
        const id = this.props.id

        const catOptions = categories.map((c, i) => {
            return <option key={i} >{c}</option>
        })

        const FeatureComponents = this.props.features.map((f, i) => {
            return (
                <Feature
                    features={this.state.catFeatures} 
                    index={i}
                    key={i}
                    selected={f.message}
                    updateFeature={this.props.updateFeature}
                    submitFeature={this.submitFeature}
                    removeFeature={this.props.removeFeature}
                />
            )
        })

        return (
            <fieldset>
                <legend>Category and Features</legend>
                <label htmlFor={`productCats-${id}`}>Category</label>
                <select 
                    id={`productCats-${id}`} 
                    onChange={e => this.updateCatFeatures(e.target.value)}
                    value={this.props.category}
                >
                    <option>Select Category</option>
                    {catOptions}
                </select>
                <p>Features</p>
                {FeatureComponents}
                <button 
                    className={styles.addFeat} 
                    onClick={e => this.props.addFeature(e)}
                >
                    Add Another Feature
                </button>
            </fieldset>
        )
    }
}