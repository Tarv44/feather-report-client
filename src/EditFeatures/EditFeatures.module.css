.addFeat {
    margin-top: 15px;
}

.select {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.buttons {
    display: flex;
    flex-direction: row;
}

.label {
    margin-bottom: 0;
}