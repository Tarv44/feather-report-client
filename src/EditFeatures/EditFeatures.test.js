import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import EditFeatures from './EditFeatures'
import { categories, features } from '../component.fixtures'

describe('----- EditFeatures -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
            <EditFeatures
                allCats={categories}
                features={features}
                category={categories[0]}
            />
        , div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <EditFeatures
                    allCats={categories}
                    features={features}
                    category={categories[0]}
                />
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})