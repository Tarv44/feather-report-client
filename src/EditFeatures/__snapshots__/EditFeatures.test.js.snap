// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- EditFeatures ----- renders the UI as expected 1`] = `
<fieldset>
  <legend>
    Category and Features
  </legend>
  <h3
    className="label"
  >
    Category
  </h3>
  <div
    className="category"
  >
    <select
      onChange={[Function]}
      value="Knives"
    >
      <option>
        Select Category
      </option>
      <option
        id={1}
      >
        Knives
      </option>
      <option
        id={2}
      >
        Blenders
      </option>
      <option
        id={3}
      >
        Toasters
      </option>
    </select>
    <button
      className="button"
      onClick={[Function]}
    >
      Create New Category
    </button>
  </div>
  <h3
    className="label"
  >
    Features
  </h3>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="8-inch"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="Ergonomic Handle"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="Carbon Stainless Steel"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="Waved Pattern"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="Pakkawood Handle"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="7-inch"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="Black"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="5-inch"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="Lifetime Warranty"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="Santoprene Handle"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="Titanium Plated"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="Comes with Sharpener"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <div
    className="feature"
  >
    <select
      onChange={[Function]}
      value="Fiery Color"
    >
      <option>
        Select Feature
      </option>
    </select>
    <button
      className="create"
      onClick={[Function]}
    >
      Create New Feature
    </button>
    <button
      onClick={[Function]}
    >
      Remove Feature
    </button>
  </div>
  <button
    className="addFeat"
    onClick={[Function]}
  >
    Add Another Feature
  </button>
</fieldset>
`;
